### PR TITLE
add a fuzz function and corpus

### PR DESCRIPTION
**Pull request by josharian**
I decided in this case to also commit the corpus, because it has been built up over several different implementations of netaddr, which means it has coverage hints that might be hard to discover given only one of those implementations.

### CODE REVIEW

File: README.md
Comment (mdlayher) [inline README.md:29]: I feel like I've had to use `CGO_ENABLED=0 go-fuzz-build` for a long time on linux/amd64 to get things working. Have you run into that?

**Review by mdlayher — APPROVED** [review]
LGTM

File: README.md
Comment (josharian) [reply README.md:29]: Oh, I see https://github.com/dvyukov/go-fuzz/issues/101. I've not run into it. 2016 is a long time ago, and I re-worked large sections of go-fuzz-build to use go/packages. I wonder whether maybe this is fixed now. (Fingers crossed!)

**Comment by josharian** [comment]
Thanks for the review. I'm going to wait on a review from Brad, because I think he may have strong feelings about checking in the corpus.

**Comment by mdlayher** [comment]
No worries. I made a separate fuzz corpus repo for my projects just to keep my code repos tidy, but I don't have strong feelings either way.

**Comment by danderson** [comment]
Adding corpus SGTM. If we want to avoid the noise of generated files in the main repo, we could do a sub-repo for the fuzzing corpus?

**Comment by bradfitz** [comment]
Can we do the corpus in a separate branch (of the same repo) instead? Then the "make fuzz" target or whatever can "git fetch" it as needed (at least if empty). I did that with storing the linux kernel for integration tests before (https://docs.google.com/presentation/d/1rAAyOTCsB8GLbMgI0CAbn69r6EVWL8j3DPl4qc0sSlc/edit#slide=id.g410de087f7_1_31) which was nice... reduced noise & download for regular users.

**Comment by josharian** [comment]
The corpus is going to be more read-write than the kernel, and it's a directory full of files. I'm more inclined to go the submodule route, if that works for you. I've created a corpus submodule and updated this PR for your perusal. WDYT?

**Review by mdlayher — APPROVED** [review]
LGTM with submodule but will leave for @bradfitz.

**Review by bradfitz — APPROVED** [review]
submodule SGTM

**Pull request merged**
